internal_workflow_testsuite.go

```diff
@@ -276,7 +277,8 @@ func (env *testWorkflowEnvironmentImpl) executeWorkflow(workflowFn interface{},
 		// auto register workflow if it is not already registered
 		fnName := getFunctionName(workflowFn)
 		if _, ok := s.hostEnv.getWorkflowFn(fnName); !ok {
-			s.RegisterWorkflow(workflowFn)
+			wo := RegisterWorkflowOptions{}
+			s.RegisterWorkflow(workflowFn, wo)
```

Comment:
I agree with your underlying point, it's nice to not have users pass in an empty struct if they don't care about the names. I think it is good practice to avoid nils though. An option we discussed is exposing two methods
RegisterWorkflowWithOptions(fn, opts)
RegisterWorkflow(fn) 

However this would mean these two sets of methods on the worker and client interfaces. We can discuss this if you think that's a good idea.